autoload/ale/engine.vim

```diff
@@ -556,6 +556,8 @@ function! s:RunJob(options) abort
         \   'output': [],
         \   'next_chain_index': l:next_chain_index,
         \}
+
+        silent doautocmd <nomodeline> User ALEStartLint
```

Comment:
`ALEStartLint` isn't consistent with the `ALELint` event here. `ALELint` is only executed after all of the jobs finish. I recommend moving this to the `RunLinters` function instead.

The question is, should the event run before jobs are run, or slightly after they are run? What if no jobs end up being run? It might be best to just run the event before any linters are run if and only if at least one linter is in the list. Then you'll be able to change options in the command before they are used.

Reply:
I changed it so that it always gets run before all jobs unless there are no defined linters. That's consistent with how `ALELint` currently works as far as I can see.